zipline/data/bundles/quandl.py

```diff
@@ -13,265 +12,140 @@
-    # cut out all the other stuff in the name column
-    # we need to escape the paren because it is actually splitting on a regex
-    data.asset_name = data.asset_name.str.split(r' \(', 1).str.get(0)
-    data['exchange'] = 'QUANDL'
-    data['auto_close_date'] = data['end_date'] + pd.Timedelta(days=1)
-    return data
+    metadata = pd.read_csv(
+        format_metadata_url(api_key)
+    )
+    # Extract link from metadata and download zip file.
+    table_url = metadata.loc[0, 'file.link']
+    if show_progress:
+        raw_file = download_with_progress(
+            table_url,
+            chunk_size=ONE_MEGABYTE,
+            label="Downloading WIKI Prices table from Quandl"
+        )
+    else:
+        raw_file = download_without_progress(table_url)
 
+    with ZipFile(raw_file) as zip_file:
+        file_names = zip_file.namelist()
+        wiki_prices = file_names.pop()
+        table_file = zip_file.open(wiki_prices)
+        if show_progress:
+            print('Parsing raw data.')
+        data_table = pd.read_csv(
+            table_file,
+            parse_dates=['date'],
+            usecols=[
+                'ticker',
+                'date',
+                'open',
+                'high',
+                'low',
+                'close',
+                'volume',
+                'ex-dividend',
+                'split_ratio'
+            ],
+            na_values=['NA']
+        ).rename(
+            columns={
+                'ticker': 'symbol',
+                'ex-dividend': 'ex_dividend'
+            }
+        )
+        table_file.close()
 
-def format_metadata_url(api_key, page_number):
-    """Build the query RL for the quandl WIKI metadata.
-    """
-    query_params = [
-        ('per_page', '100'),
-        ('sort_by', 'id'),
-        ('page', str(page_number)),
-        ('database_code', 'WIKI'),
-    ]
-    if api_key is not None:
-        query_params = [('api_key', api_key)] + query_params
-    return (
-        'https://www.quandl.com/api/v3/datasets.csv?' + urlencode(query_params)
-    )
+    return data_table
 
 
-def format_wiki_url(api_key, symbol, start_date, end_date):
-    """
-    Build a query URL for a quandl WIKI dataset.
-    """
-    query_params = [
-        ('start_date', start_date.strftime('%Y-%m-%d')),
-        ('end_date', end_date.strftime('%Y-%m-%d')),
-        ('order', 'asc'),
-    ]
-    if api_key is not None:
-        query_params = [('api_key', api_key)] + query_params
+def gen_asset_metadata(data, show_progress):
+    if show_progress:
+        print('Generating asset metadata.')
 
-    return (
-        "https://www.quandl.com/api/v3/datasets/WIKI/"
-        "{symbol}.csv?{query}".format(
-            symbol=symbol,
-            query=urlencode(query_params),
-        )
-    )
+    symbols = data['symbol'].unique()
+    asset_metadata = {}
 
+    asset_metadata['symbol'] = symbols
+    data.set_index('symbol', inplace=True)
+    asset_metadata['start_date'] = \
```

Comment:
This might be something that can be replaced a `groupby` on a DataFrame, something like:

```
asset_metadata = (data.ix[:, ['symbol', 'date']]
           .groupby(by='symbol',)
           .agg({'date': [np.min, np.max]})
           .reset_index())
asset_metadata['start_date'] = asset_metadata.date.amin
asset_metadata['end_date'] = asset_metadata.date.amax
del asset_metadata['date']
asset_metadata.columns = asset_metadata.columns.get_level_values(0)
```